Corrected filename and include static types

NumberPiOso opened the pull request:
continuing in line with #2128.

My first time contributing code to open source so criticism is encouraged.

- The name of the file is now compliant with python naming conventions
- Add static type as stated in contributing guidelines

### **Describe your change:**

* [x] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.

spamegg1 commented:
Hi @NumberPiOso ,
It says you deleted the file, and rewrote it from scratch? It would have been better if you just edited it line by line so we could easily see the differences under "Files changed."

Also I see that you included type hints for some variables too. Like:
```python
user_input: str = input("Enter numbers separated by comma:\n").strip()
sequence: List[int] = [int(item.strip()) for item in user_input.split(",")]
```
etc.

I think [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md) says to provide type hints only for the input parameters and return values of functions and nothing else, because it causes too much visual clutter.

Good luck and happy coding!

cclauss commented on searches/simple_binary_search.py:
```diff
@@ -0,0 +1,45 @@
+Pure Python implementation of the binary search algorithm.
+
+For doctests run following command:
+python -m doctest -v simple_binary_search.py
```
```suggestion
```

cclauss commented on searches/simple_binary_search.py:
```diff
@@ -0,0 +1,45 @@
+
+For doctests run following command:
+python -m doctest -v simple_binary_search.py
+or
```
```suggestion
```

cclauss commented on searches/simple_binary_search.py:
```diff
@@ -0,0 +1,45 @@
+python3 -m doctest -v simple_binary_search.py
+
+For manual testing run:
+python simple_binary_search.py
```
```suggestion
python3 simple_binary_search.py
```

cclauss commented on searches/simple_binary_search.py:
```diff
@@ -0,0 +1,45 @@
+
+def binary_search(a_list: List[int], item: int) -> bool:
+    """
+    Pure python implementation of a binary search of a number is in a list.
```
```suggestion
```
This was said already.

cclauss commented on searches/simple_binary_search.py:
```diff
@@ -0,0 +1,45 @@
+    >>> print(binary_search(test_list, 3))
+    False
+    >>> print(binary_search(test_list, 13))
+    True
```
Please add tests for negative integers, floats, repeated numbers, letters.

cclauss commented on searches/simple_binary_search.py:
```diff
@@ -0,0 +1,45 @@
+
+
+if __name__ == "__main__":
+    user_input: str = input("Enter numbers separated by comma:\n").strip()
```
```suggestion
    user_input = input("Enter numbers separated by comma:\n").strip()
```
Python knows that input() returns a str.

cclauss commented on searches/simple_binary_search.py:
```diff
@@ -0,0 +1,45 @@
+
+if __name__ == "__main__":
+    user_input: str = input("Enter numbers separated by comma:\n").strip()
+    sequence: List[int] = [int(item.strip()) for item in user_input.split(",")]
```
```suggestion
    sequence = [int(item.strip()) for item in user_input.split(",")]
```

cclauss commented on searches/simple_binary_search.py:
```diff
@@ -0,0 +1,45 @@
+    user_input: str = input("Enter numbers separated by comma:\n").strip()
+    sequence: List[int] = [int(item.strip()) for item in user_input.split(",")]
+    target_str = input("Enter the number to be found in the list:\n").strip()
+    target: int = int(target_str)
```
```suggestion
    target = int(target_str)
```

NumberPiOso commented:
Hi @spamegg1 
> It says you deleted the file, and rewrote it from scratch? It would have been better if you just edited it line by line so we could easily see the differences under "Files changed."

It is because the other file was named **simple-binary-search.py** so I renamed it to **simple_binary_search.py**

I did it using *git mv*, is there any other way to do this in order for github to show us the differences?

> Also I see that you included type hints for some variables too. Like:
> 
> ```python
> user_input: str = input("Enter numbers separated by comma:\n").strip()
> sequence: List[int] = [int(item.strip()) for item in user_input.split(",")]
> ```
> 
> etc.
> 
> I think [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md) says to provide type hints only for the input parameters and return values of functions and nothing else, because it causes too much visual clutter and Python is smart enough to figure it out. (There might be some exceptions to this but I don't know.)
> 

Totally agreed, I will correct it.

Thank you.

cclauss reviewed (approved):
LGTM.  It would be easier to review as two PRs.  One that changed the filename and the other that changed the file contents.
